### PR TITLE
Add concept of global resources which can be shared across applications

### PR DESCRIPTION
**Pull request by lceleban-dd**

### CODE REVIEW

File: akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala [hunk 1]
```diff
@@ -34,6 +35,14 @@ private[akkeeper] final class AkkeeperConfig(akkeeperConfig: Config) {
       Seq.empty
     }
   }
+
+  lazy val globalResources: Seq[AkkeeperResource] = {
+    if (akkeeperConfig.hasPath("globalResources")) {
+      akkeeperConfig.getConfigList("globalResources").asScala.map(akkeeperResourceFromConfig)
```
Comment (izeigerman) [inline akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala]: Can we please name this `global-resources` to be consistent with other property names.

File: akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala [hunk 2]
```diff
@@ -120,4 +129,12 @@ object ConfigUtils {
       environment = config.getMapOfStrings("environment")
     )
   }
+
+  private[config] def akkeeperResourceFromConfig(config: Config): AkkeeperResource = {
+    AkkeeperResource(
+      new URI(config.getString("uri")),
+      config.getString("localPath"),
```
Comment (izeigerman) [inline akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala]: Same as above: `local-path`.

File: akkeeper-yarn/src/main/scala/akkeeper/yarn/YarnLocalResourceManager.scala [hunk 3]
```diff
@@ -55,11 +58,19 @@ private[akkeeper] final class YarnLocalResourceManager(conf: Configuration,
...
+    val dstStatus = dstFs.getFileStatus(dstPath)
+    create(dstFs, dstStatus, localResourceType, localResourceVisibility)
+  }
+
   def createLocalResource(srcStream: InputStream, dstPath: String): LocalResource = {
-    val dstFs = stagingDirPath.getFileSystem(conf)
-    val dst = copyResourceToStagingDir(dstFs, srcStream, dstPath)
-    val dstStatus = dstFs.getFileStatus(dst)
-    create(dstFs, dstStatus)
+    val dst = copyResourceToStagingDir(srcStream, dstPath)
```
Comment (izeigerman) [inline akkeeper-yarn/src/main/scala/akkeeper/yarn/YarnLocalResourceManager.scala]: [Thought|Optional] Perhaps it makes sense to rename all `createLocalResource` methods to `uploadLocalResource` to draw a clear line between the case when we copy the resource and when we use the original source instead.

File: akkeeper-yarn/src/main/scala/akkeeper/yarn/LocalResourceNames.scala [hunk 4]
```diff
@@ -19,6 +19,7 @@ private[akkeeper] object LocalResourceNames {
   val AkkeeperJarName = "akkeeper.jar"
   val UserJarName = "user.jar"
   val UserConfigName = "user_config.conf"
+  val LauncherConfigName = "launcher_config.conf"
```
Comment (izeigerman) [inline akkeeper-yarn/src/main/scala/akkeeper/yarn/LocalResourceNames.scala]: I wouldn't say it's a launcher config per se. Morel like `application` or `akkeeper` config.

File: akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala (hunk 1 above)
Comment (lceleban-dd) [reply akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala]: Yes, done

File: akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala (hunk 2 above)
Comment (lceleban-dd) [reply akkeeper-common/src/main/scala/akkeeper/common/config/Configs.scala]: Done

File: akkeeper-yarn/src/main/scala/akkeeper/yarn/YarnLocalResourceManager.scala (hunk 3 above)
Comment (lceleban-dd) [reply akkeeper-yarn/src/main/scala/akkeeper/yarn/YarnLocalResourceManager.scala]: Renamed

File: akkeeper-yarn/src/main/scala/akkeeper/yarn/LocalResourceNames.scala (hunk 4 above)
Comment (lceleban-dd) [reply akkeeper-yarn/src/main/scala/akkeeper/yarn/LocalResourceNames.scala]: Renamed to `application.conf`

File: akkeeper-yarn/src/main/scala/akkeeper/yarn/LocalResourceNames.scala [hunk 5]
```diff
@@ -19,6 +19,7 @@ private[akkeeper] object LocalResourceNames {
   val AkkeeperJarName = "akkeeper.jar"
   val UserJarName = "user.jar"
   val UserConfigName = "user_config.conf"
+  val LauncherConfigName = "application.conf"
```
Comment (izeigerman) [inline akkeeper-yarn/src/main/scala/akkeeper/yarn/LocalResourceNames.scala]: Shall we rename the constant name as well?

Comment (lceleban-dd) [reply akkeeper-yarn/src/main/scala/akkeeper/yarn/LocalResourceNames.scala]: :man_facepalming: :wink:

done

**Pull request merged**